src/utils.rs

```diff
@@ -39,6 +39,18 @@ pub(crate) fn digits(mut number: usize, mut basis: u8) -> u32 {
     digits
 }
 
+/// Helper function to create an int slice from a pointer and size
+/// which special cases when cnt is 0 (in which case c_int may be null)
+/// UNSAFE: The slice returned has 'static' lifetime, as the lifespan is based
+/// on the input pointer and size.
+pub unsafe fn from_raw_parts_maybe_null<T>(ptr: *const T, cnt: usize) -> &'static [T] {
```

Comment:
```suggestion
pub unsafe fn from_raw_parts_maybe_null<'a, T>(ptr: *const T, cnt: usize) -> &'a [T] {
```

I think it should be possible to use the same signature the `std::slice::from_raw_parts` uses, but I'm writing this without checking against the compiler right now.